src/app/config/confparse.c

```diff
@@ -550,13 +550,52 @@ config_var_is_gettable(const config_var_t *var)
   return true;
 }
 
-bool
-config_var_is_contained(const config_var_t *var)
+/**
+ * Return true iff this is variable is "derived" from another -- that is,
+ * inspecting this variable inspects part of another, and changing this
+ * variable changes part of another.
+ *
+ * Derived variables require special handling in several ways: they do not
+ * need to be copied independently when we are copying a config object, since
+ * copying the variable they are derived from copies them too.  Similarly,
+ * they do not need to be compared independently when listing changes, since
+ * comparing the variable that they are derived from compares them too.
+ **/
+static bool
+config_var_is_derived(const config_var_t *var)
 {
   return struct_var_is_contained(&var->member);
 }
 
 /**
+ * Return true iff we need to check <b>var</b> for changes when we are
+ * comparing config options for changes.
+ *
+ * A false result might mean that the variable is a derived variable, and that
+ * comparing the variable it derives from compares this one too-- or it might
+ * mean that there is no data to compare.
```

Comment:
Shouldn't we be checking obsolete here, too?

Reply:
After the refactor, obsolete is no longer a flag.

(Obsolete falls under the category of variables that have no data to copy.)

Reply:
"Obsolete" is supposed to count as a kind of "derived", which is one of the reasons I eliminated the "derived" notion.  By the end of the branch, we are checking the NOCOPY and NOCMP flags for these functions: obsolete things have those.